src/include/sof/dai.h

```diff
@@ -102,7 +104,7 @@ struct dai_plat_data {
 	uint32_t base;
 	uint32_t irq;
 	uint32_t flags;
-	struct dai_plat_fifo_data fifo[2];
+	struct dai_plat_fifo_data *fifo;
```

Comment:
I am not sure this is a required change. this leads to a lot of changes for SSP/DMIC/HDA, will increase code size for no good reason, and you aren't using this field at all for ALH. So why not keep the fifo[2] and just implement get_fifo/handshake in a ALH-specific way when you need them? You would waste a bit of memory that's not used for ALH but it'd be more efficient in the end, no?

Reply:
This was my thought, initially I thought about allocating a larger array here for ALH, but I wasn't the fan having such large structures on stack, hence this. So yeah, I will revert to using 2 default fifo structures.